server/src/main/java/com/cloud/user/AccountManagerImpl.java

```diff
@@ -1054,6 +1064,9 @@ public UserAccount createUserAccount(final String userName, final String passwor
         // Check permissions
         checkAccess(getCurrentCallingAccount(), domain);
 
+//        get user by account
```

Comment:
```suggestion
        // Check role escalation permission for the account
```

remove comment if not required

Reply:
work in progress @sureshanaparti . I need to keep options open atm.